infrastructure/demo/main/gh.tf

```
@@ -0,0 +1,105 @@
+# Github OICD provider
+data "aws_iam_openid_connect_provider" "github" {
+  url = "https://token.actions.githubusercontent.com"
+}
+
+data "aws_iam_policy_document" "github_actions_assume_role" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    principals {
+      type        = "Federated"
+      identifiers = [data.aws_iam_openid_connect_provider.github.arn]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "token.actions.githubusercontent.com:aud"
+      values   = ["sts.amazonaws.com"]
+    }
+    condition {
+      test     = "StringLike"
+      variable = "token.actions.githubusercontent.com:sub"
+      values = [
+        "repo:IDinsight/aaq-core:*"
+      ]
+    }
+  }
+
+}
+
+resource "aws_iam_role" "github_actions" {
+  name               = "github-actions"
+  assume_role_policy = data.aws_iam_policy_document.github_actions_assume_role.json
+}
```

This will clash if I try to create infra for another environment or another project..

1. Should we have separate GA role for separate projects (repositories)?
2. Should we have separate GA role for separate environments? This I'm not sure about. Probably not? 
But then where to we define and create project-wide github actions role? Some folder like `infrastructure/common` where we delcare common resources for a given project?

If yes to both, we should change this to 
```suggestion
resource "aws_iam_role" "github_actions" {
  name               = "${var.project_name}-${var.environment}-github-actions"
  assume_role_policy = data.aws_iam_policy_document.github_actions_assume_role.json
}
```
but not confident answer is yes to both!

---

We don't have to address it now but would be nice to get your thoughts on the direction!